addons/hr_timesheet/models/project_project.py

```diff
@@ -137,10 +137,10 @@ def _compute_total_timesheet_time(self):
             # if the timesheet has no product_uom_id then we take the one of the project
             total_time = 0.0
             for product_uom, unit_amount in timesheet_time_dict[project.id]:
-                factor = (product_uom or project.timesheet_encode_uom_id).factor_inv
+                factor = (product_uom or project.timesheet_encode_uom_id).factor
                 total_time += unit_amount * (1.0 if project.encode_uom_in_days else factor)
             # Now convert to the proper unit of measure set in the settings
-            total_time *= project.timesheet_encode_uom_id.factor
+            total_time /= project.timesheet_encode_uom_id.factor
```

Comment:
Ah yeah, this is where the conversion might be wrong if you don't change the reference uom in the upgrade script :crying_cat_face:

Reply:
Yes, indeed. I will handle this in the upgrade script.